cmd/thanos/rule.go

```diff
@@ -521,31 +574,32 @@ func runRule(
 	)
 
 	// Start gRPC server.
-	{
-		tsdbStore := store.NewTSDBStore(logger, db, component.Rule, conf.lset)
+	tlsCfg, err := tls.NewServerConfig(log.With(logger, "protocol", "gRPC"), conf.grpc.tlsSrvCert, conf.grpc.tlsSrvKey, conf.grpc.tlsSrvClientCA)
+	if err != nil {
+		return errors.Wrap(err, "setup gRPC server")
+	}
 
-		tlsCfg, err := tls.NewServerConfig(log.With(logger, "protocol", "gRPC"), conf.grpc.tlsSrvCert, conf.grpc.tlsSrvKey, conf.grpc.tlsSrvClientCA)
-		if err != nil {
-			return errors.Wrap(err, "setup gRPC server")
-		}
+	options := []grpcserver.Option{
+		grpcserver.WithServer(thanosrules.RegisterRulesServer(ruleMgr)),
+		grpcserver.WithListen(conf.grpc.bindAddress),
+		grpcserver.WithGracePeriod(time.Duration(conf.grpc.gracePeriod)),
+		grpcserver.WithTLSConfig(tlsCfg),
+	}
+	if db != nil {
+		tsdbStore := store.NewTSDBStore(logger, db, component.Rule, conf.lset)
+		options = append(options, grpcserver.WithServer(store.RegisterStoreServer(tsdbStore)))
+	}
+	// TODO: Add rules API implementation when ready.
```

Comment:
I think this is an outdated comment that we can remove since we have the Rules server?

Reply:
Done